R/weight.R

```diff
@@ -68,6 +89,49 @@ setMethod("weightVariables", "VariableCatalog", function (x) {
     }
 })
 
+#' @rdname weightVariables
+#' @export
+`weightVariables<-` <- function(x, value) {
+    if (is.dataset(x)) {
+        varCat <- allVariables(x)
+    } else if (inherits(x,  "VariableCatalog")) {
+        varCat <- x
+    } else {
+        halt("Left hand side is not a Crunch Dataset or Variable Catalog.")
+    }
+    current <- crGET(shojiURL(varCat, "orders", "weights"))
+    new <- current
+    if (is.null(value)) {
+        new$graph <- NULL
+    } else {
+        if (is.variable(value) || length(value) == 1) {
+            value <- list(value)
+        }
+        var_names <- vapply(value, function(x){
+            if (is.variable(x)) {
+                return(name(x))
+            } else {
+                return(as.character(x))
+            }
+            }, character(1))
+        all_var <- vapply(value, is.Numeric, logical(1))
+        if (!all(all_var)) {
+            err_text <- " is not a numeric Crunch variable."
+            if ( sum(!all_var) > 1) {
+                err_text <- " are not a numeric Crunch variables."
+            }
+            halt( paste0(serialPaste(var_names[!all_var]), err_text))
+        }
+        new$graph <- lapply(value, self)
+    }
+    if (!identical(current, new)) {
+        crPATCH(shojiURL(varCat, "orders", "weights"),
```

Comment:
Order resources should take a PUT instead of a PATCH because you're overwriting the entire resource. PATCH is poorly defined.